src/content/en/2022/javascript.md

```diff
@@ -19,4 +19,862 @@ featured_stat_label_3: TODO
+
+{{ figure_markup(
+    image="uncompressed.png",
+    caption="Uncompressed resources by size.",
+    description="TODO",
+    chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTok8UGqYaA58uKaASB2pfM2jlmQu6g3kwHxB6Lb8L7dbccyDtQht823YhchdXRG8SZpB3asRayJI97/pubchart?oid=1991115497&format=interactive",
+    sheets_gid="1609356066",
+    sql_file="compression_none_by_bytes.sql"
+  )
+}}
+
+Thankfully, we're seeing that it's only mostly the smallest resources, specifically those third-party scripts that have payloads smaller than 5 KB, that are being delivered without compression. This is because compression yields diminishing returns when applied to small resources, and in fact, the added overhead of dynamic compression may cause delayed resource delivery. There are, unfortunately, some opportunities across the spectrum to compress larger resources, such as some first-party scripts with payloads over 100 KB.
+
+_Always_ check your compression settings to ensure you're delivering the smallest possible script payloads over the network, and remember: compression speeds up resource _delivery_. Those scripts, once delivered to the browser, will be decompressed and their processing time will not change due to compression. Compression is not a good excuse to deliver huge script payloads that can make interactivity worse during startup.
+
+### Minification
+
+Minification of text assets is a time-tested practice for reducing file size. The practice involves removing all of the unnecessary spaces and comments from source code in order to reduce their transfer size. A further step known as uglification is applied to JavaScript, which reduces all of the variables, class names, and function names in a script to shorter, unreadable symbols. Lighthouse's [Minify JavaScript](https://web.dev/unminified-javascript/) audit checks for unminified JavaScript.
+
+{{ figure_markup(
+    image="lighthouse-unminified.png",
+    caption="Distribution of unminified JavaScript audit scores.",
+    description="TODO",
+    chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTok8UGqYaA58uKaASB2pfM2jlmQu6g3kwHxB6Lb8L7dbccyDtQht823YhchdXRG8SZpB3asRayJI97/pubchart?oid=968128936&format=interactive",
+    sheets_gid="906447348",
+    sql_file="lighthouse_unminified_js.sql"
+  )
+}}
+
+Here, 0.00 represents the worst score whereas 1.00 represents the best score. 68% of mobile pages are scoring between 0.9 and 1.0 on Lighthouse's minified JavaScript audit, whereas the figure for desktop pages is 79%. This means that on mobile, 32% of pages have opportunities to ship minified JavaScript, whereas that figure for desktop pages is 21%.
+
+{# TODO(editors): This metric corresponds to the 3P chapter, not the JS chapter. Edit this to ensure that the metadata links to the right source sheet/query. #}
```

Comment:
@tunetheweb what's the way to reference a metric from another chapter?

Reply:
You can insert it using relative paths (including the full sheets gid URL) like this:

https://github.com/HTTPArchive/almanac.httparchive.org/blob/f66e3ff3147fd4b5f02d3b42c327716a183e41d1/src/content/en/2020/javascript.md#L28-L35

Personally my preference is to duplicate the query and data in this chapter and present a slightly different spin on it rather than just duplicate it. A bit of repetition but removes dependency on another chapter, and allows this chapter name to be on the graph.